### ISSUE TITLE
implementation of fGES (fast greedy equivalence search)

### ISSUE THREAD

**Issue by huiouyang16** [cmu-phil/causal-learn#84]
Does this repo have plans to implement the algorithm fGES[1]? fGES seems to work well for large scale problems. I wanna do some work on a large scale problem. If there is a related plan, it will help to use fGES more conveniently on the python platform, instead of calling Tetrad implemented in Java.

[1] Ramsey J, Glymour M, Sanchez-Romero R, et al. A million variables and more: the fast greedy equivalence search algorithm for learning high-dimensional graphical causal models, with an application to functional magnetic resonance images[J]. International journal of data science and analytics, 2017, 3(2): 121-129.

**Comment by kunwuz** [comment]
Thanks for reaching out. Yes, we do have the plan to implement fGES on python, but it's still in its early stage. Please let me know if you (or/and anyone) would like to contribute. We are more than grateful for any help that benefits the community :)

**Comment by huiouyang16** [comment]
> Thanks for reaching out. Yes, we do have the plan to implement fGES on python, but it's still in its early stage. Please let me know if you (or/and anyone) would like to contribute. We are more than grateful for any help that benefits the community :)

Yeah, of course I would like to do some related work. But I'm not sure if it can be done well since my current knowledge of fGES is limited. I'm willing to give it a shot, but may need some help along the way.

**Comment by cianeastwood** [comment]
Any update on this?

**Comment by kunwuz** [comment]
Unfortunately, we still do not have our own implementation of fGES in causal-learn. But if helpful, here is another Python implementation of the algorithm: [https://github.com/eberharf/fges-py](https://github.com/eberharf/fges-py)

**Comment by Jessez2** [comment]
Any update on this?
